hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/map/AbstractMapQueryMessageTask.java

```diff
@@ -289,4 +295,18 @@ private MapOperation createQueryPartitionOperation(Query query, Version clusterV
             return new QueryPartitionOperation(query);
         }
     }
+
+    private void assertAllPartitionsQueried(BitSet finishedPartitions, int partitionCount) {
+        if (hasMissingPartitions(finishedPartitions, partitionCount)) {
+            int missedPartitionsCount = 0;
+            for (int i = 0; i < partitionCount; i++) {
+                if (!finishedPartitions.get(i)) {
+                    missedPartitionsCount++;
+                }
+            }
+            throw rethrow(new QueryException("Query aborted. Could not execute query for all partitions. Missed "
```

Comment:
`QueryException` is already a `RuntimeException` so no need to `rethrow(..)`

Reply:
Yeah, I just don't know all the rules about Codecs for exceptions and this peel method. So you are sure I can remove it? Cc @asimarslan

Reply:
Agree to @vbekiaris . 
Moreover QueryException is defined on protocol. You can send it to client safely